GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -259,30 +263,20 @@ public bool IsReadyForExternalAcquireLockRequests(NamedPipeMessages.LockData req
         public EventMetadata GetMetadataForHeartBeat(ref EventLevel eventLevel)
         {
             EventMetadata metadata = new EventMetadata();
-            if (this.placeHolderCreationCount.Count > 0)
-            {
-                ConcurrentDictionary<string, PlaceHolderCreateCounter> collectedData = this.placeHolderCreationCount;
-                this.placeHolderCreationCount = new ConcurrentDictionary<string, PlaceHolderCreateCounter>(StringComparer.OrdinalIgnoreCase);
-
-                int count = 0;
-                foreach (KeyValuePair<string, PlaceHolderCreateCounter> processCount in
-                    collectedData.OrderByDescending((KeyValuePair<string, PlaceHolderCreateCounter> kvp) => kvp.Value.Count))
-                {
-                    ++count;
-                    if (count > 10)
-                    {
-                        break;
-                    }
-
-                    metadata.Add("ProcessName" + count, processCount.Key);
-                    metadata.Add("ProcessCount" + count, processCount.Value.Count);
-                }
-
-                eventLevel = EventLevel.Informational;
-            }
+            metadata.Add(
+                "FilePlaceholderCreation",
+                this.GetProcessInteractionData(ref this.filePlaceHolderCreationCount, ref eventLevel));
+             metadata.Add(
+                "FolderPlaceholderCreation",
+                this.GetProcessInteractionData(ref this.folderPlaceHolderCreationCount, ref eventLevel));
+            metadata.Add(
+                "FilePlaceholdersHydrated",
+                this.GetProcessInteractionData(ref this.fileHydrationCount, ref eventLevel));
 
             metadata.Add("ModifiedPathsCount", this.modifiedPaths.Count);
-            metadata.Add("PlaceholderCount", this.placeholderDatabase.GetCount());
+            metadata.Add("FilePlaceholderCount", this.placeholderDatabase.GetFilePlaceholdersCount());
+            metadata.Add("FolderPlaceholderCount", this.placeholderDatabase.GetFolderPlaceholdersCount());
+

```

Comment:
GetExpandedFolderCount? too?

Reply:
I don't see where GetExpandedFolderCount is used, if we don't need it maybe drop it?

Reply:
Good catch, I thought this would be interesting data to have, but upon further thought, I am not going to use it.